hacker_Rank/May4_Answer.py

```diff
@@ -0,0 +1,33 @@
+def array_rotation(a, k):
+    result = []
+    print(a)
+    step = 1
+    for i in range(k):
+        step = 1
+        print("for_loop:" + str(i), a)
+        while step < len(a):
+            temp = a[0]
+            a[0] = a[(0+step)]
+            a[(0+step)] = temp
+            print("while_loop:" + str(step) , a)
```

Comment:
Can you explain me this logic ?

Reply:
step is nothing but carry forward of the variables . we required total (len(a)-1) step 
ex: a=[1,2,3,4,5]
step1 = a=[2,1,3,4,5]
step2 = a=[3,1,2,4,5]
step3 = a=[4,1,2,3,5]
step4 = a=[5,1,2,3,4]